test/integration/gateway_test.go

```diff
@@ -236,56 +236,9 @@ func TestGatewayListenerConflicts(t *testing.T) {
 				}
 			}
 		}
-		return !badhttpReady && badhttpConflicted && !badudpReady && badudpConflicted && !httpReady && httpConflicted
+		return !badudpReady && badudpConflicted && !httpReady && httpConflicted
 	}, gatewayUpdateWaitTime, time.Second)
 
-	t.Log("changing listeners to a set with conflicting hostnames")
-	// these both use the empty hostname
-	gw.Spec.Listeners = []gatewayv1beta1.Listener{
-		{
-			Name:     "httpsalpha",
-			Protocol: gatewayv1beta1.HTTPSProtocolType,
-			Port:     gatewayv1beta1.PortNumber(443),
-		},
-		{
-			Name:     "httpsbravo",
-			Protocol: gatewayv1beta1.HTTPSProtocolType,
-			Port:     gatewayv1beta1.PortNumber(443),
-		},
-	}
-
-	_, err = gatewayClient.GatewayV1beta1().Gateways(ns.Name).Update(ctx, gw, metav1.UpdateOptions{})
-	require.NoError(t, err)
-
-	t.Log("confirming listeners with conflicted hostnames receive appropriate conditions")
```

Comment:
Is this going to be re-added when a fix for https://github.com/Kong/kubernetes-ingress-controller/issues/4597 gets implemented?

Reply:
Things that will get re-added are commented out and marked with a TODO link to that PR.

Most of the conflicts caught by CEL (including this one) are in clear violation of the spec and should be rejected up front. We no longer need to (and indeed can't) test them independently since they're just rejected by the API server.